File: year2/semestrul2/MPP/Laboratoare/lab3_cscs/lab3_cscs/Repository/iRepository.cs
using System.Collections.Generic;
using lab3_cscs.Model;

namespace lab3_cscs.Repository
{
    /// <summary>
    /// CRUD operations repository interface
    /// </summary>
    /// <typeparam name="ID">Type E must have an attribute of type ID</typeparam>
    /// <typeparam name="E">Type of entities saved in repository</typeparam>
    public interface IRepository<ID, E> where E : Entity<ID>
    {
        E? FindOne(ID id);
        /// <summary>
        /// @return all entities
        /// </summary>
        IEnumerable<E> FindAll();
        E? Save(E entity);
        E? Delete(ID id);
        E? Update(E entity);
    }
}

File: year2/semestrul2/MPP/Laboratoare/lab4recuperare/src/main/java/org/example/Repository/Interfaces/iRepository.java
package org.example.Repository.Interfaces;
import org.example.Model.Entity;
import org.example.Model.Inscriere;

import java.util.Optional;

/**
 * CRUD operations repository interface
 * @param <ID> - type E must have an attribute of type ID
 * @param <E> - type of entities saved in repository
 */
public interface iRepository<ID,E extends Entity<ID>> {
    Optional<Inscriere> findOne(ID id);
    /**
     *
     * @return all entities
     */
    Iterable<E> findAll();
    Optional<Inscriere> save(E entity);
    Optional<Inscriere> delete(ID id);
    E update(ID id,E entity);
}